sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataTypeConversions.scala

```diff
@@ -20,24 +20,21 @@ package org.apache.spark.sql.types
 import java.text.SimpleDateFormat
 
 import org.apache.spark.sql.Row
-import org.apache.spark.sql.catalyst.ScalaReflection
-import org.apache.spark.sql.catalyst.expressions.GenericMutableRow
+import org.apache.spark.sql.catalyst.CatalystTypeConverters
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
 
 
 private[sql] object DataTypeConversions {
 
   def productToRow(product: Product, schema: StructType): Row = {
```

Comment:
I replaced the usage inside LocalRelation with the new stuff, so now this method isn't used anywhere. Should we leave it around with a warning that it's slow, or ask people to use converters at all times?